src/main/java/dev/morling/onebrc/CalculateAverage_abeobk.java

```diff
@@ -330,13 +330,16 @@ public static void main(String[] args) throws InterruptedException, IOException
             }
 
             if (SHOW_ANALYSIS) {
-                System.out.println("total=" + Arrays.stream(lenhist).sum());
-                System.out.println("length_histogram = "
+                debug("Collision stat: ");
```

Comment:
Just to be sure: does your implementation guarantee to handle collisions 100% correctly? Or can there be a (however small) risk for unhandled collisions and thus incorrect results? I suppose it's the former.

Reply:
Yes it is handled correctly (with the assumptions given in the 1brc description).
Since there are only10k stations at most, the hash table size is intentionally added 10000, so there is no need to mod the table size.  The performance gains was achieved by using a fast/low collision mixer ( 0 collision for standard set, still low for 10k set, I had some fun tweaking this function :) )  and by removing lots of unnecessary branching.

Reply:
Got it, thx.